src/core/actions/cityactions/ResourceGathering.java

```diff
@@ -23,20 +24,69 @@ public ResourceGathering(City c)
 
     @Override
     public LinkedList<Action> computeActionVariants(final GameState gs) {
-        //TODO: compute resource gathering
-        return null;
+        LinkedList<Action> resources = new LinkedList<>();
+        if(isFeasible(gs)){
+            ResourceGathering resourceGathering = new ResourceGathering(this.city);
+            resourceGathering.setResource(this.resource);
+            resources.add(resourceGathering);
+        }
+        return resources;
     }
 
     @Override
     public boolean isFeasible(final GameState gs)
     {
-        //TODO: isFeasible ResourceGathering
+        Board b = gs.getBoard();
+        // Check if resource in range
+        for(int x = this.city.getX()- this.city.getBound(); x < x+ this.city.getBound(); x++){
+            for(int y = this.city.getX()- this.city.getBound(); y < y+ this.city.getBound(); y++){
+                if(b.getResourceAt(x,y) == this.resource){
+                    switch (this.resource){
+                        case FISH:
+                            if(b.getTribe(this.city.getTribeId()).getTechTree().isResearched(Types.TECHNOLOGY.FISHING))
+                                return true;
+                            else
+                                return false;
+                        case ORE:
+                            if(b.getTribe(this.city.getTribeId()).getTechTree().isResearched(Types.TECHNOLOGY.MINING))
+                                return true;
+                            else
+                                return false;
+                            case WHALES:
+                            if(b.getTribe(this.city.getTribeId()).getTechTree().isResearched(Types.TECHNOLOGY.WHALING))
+                                return true;
+                            else
+                                return false;
+                    }
+                }
+                    return true;
+                }
+            }
         return false;
     }
 
+
     @Override
     public boolean execute(GameState gs) {
-        //TODO: execute resource gathering
+        //Check if action feasible before execution
+        if(isFeasible(gs)){
+            switch (this.resource){
+                case CROPS:
+                case FISH:
+                case ANIMAL:
+                case FRUIT:
+                    this.city.addExtraStar(2);
```

Comment:
Gathering resources add population, not stars, right? https://github.com/GAIGResearch/Tribes/wiki/Actions